Scrapping/Foodland Scraper/scraper_foodland.py

```diff
@@ -0,0 +1,191 @@
+# ======IMPORTS======
+import requests
+from bs4 import BeautifulSoup
+import pandas as pd
+import json
+from pymongo.mongo_client import MongoClient
+from pymongo.server_api import ServerApi
+from datetime import datetime
+from urllib.parse import quote_plus
+import os
+from dotenv import load_dotenv
+
+# =====HEADERS======
+headers = {
+    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'
+}
+
+# ====== FUNCTIONS =====
+def get_text_or_default(value, default="N/A"):
+    if value:
+        return value.text.strip()
+    return default
+
+def get_total_pages(soup):
+    pagination = soup.find('div', class_='mfl-pagination')
+    if pagination:
+        page_links = pagination.find_all('a')
+        page_numbers = [int(link.get_text()) for link in page_links if link.get_text().isdigit()]
+        return max(page_numbers, default=1)
+    return 1
+
+def get_total_pages_adelaide(soup):
+    page_links = soup.find_all(class_='pagination__item__link')
+    if page_links:
+        page_numbers = [int(link.get_text()) for link in page_links if link.get_text().isdigit()]
+        return max(page_numbers, default=1)
+    return 1
+
+# Setup MongoDB connection
+def setup_mongo():
+    # Load environment variables from .env file if it exists
+    load_dotenv()
+    
+    # MongoDB credentials
+    username = os.getenv('MONGO_USERNAME', 'discountmate')
+    password = os.getenv('MONGO_PASSWORD', 'discountmate1')
```

Comment:
Please remove the actual username and password

Reply:
Regardless I have sent a new version throguh